az-digital/az_quickstart
# Close #3033 Ensure AZ Publication is reinstallable after uninstallation.

## Description
Separating this from https://github.com/az-digital/az_quickstart/pull/2574 since it is unrelated. I made these changes while developing a new feature for az_publication module after I found out I couldn't test my installation without either:

Running the following command to wipe the database, reinstall quickstart, then install publication, demo and devel modules.

```
lando install && lando drush en -y az_publication az_demo devel 
```

Or via `drush config:delete`

```
lando drush pm:uninstall -y az_publication
lando drush config:delete -y  auto_entitylabel.settings.az_author.az_author az_publication.az_citation_style.az_ama az_publication.az_citation_style.az_apa az_publication.az_citation_style.az_bluebook az_publication.az_citation_style.az_chicago az_publication.az_citation_style.az_ieee az_publication.az_citation_style.az_mla az_publication.settings core.base_field_override.node.az_publication.promote core.entity_form_display.az_author.az_author.default core.entity_form_display.node.az_publication.default core.entity_view_display.az_author.az_author.default core.entity_view_display.node.az_publication.az_card core.entity_view_display.node.az_publication.az_publication_reference core.entity_view_display.node.az_publication.default core.entity_view_display.node.az_publication.teaser core.entity_view_mode.node.az_publication_reference field.field.az_author.az_author.field_az_author_drop_particle field.field.az_author.az_author.field_az_author_fname field.field.az_author.az_author.field_az_author_link field.field.az_author.az_author.field_az_author_literal field.field.az_author.az_author.field_az_author_lname field.field.az_author.az_author.field_az_author_nondrop_particle field.field.az_author.az_author.field_az_author_parse field.field.az_author.az_author.field_az_author_person field.field.az_author.az_author.field_az_author_suffix field.field.node.az_publication.field_az_accessed_date field.field.node.az_publication.field_az_authors field.field.node.az_publication.field_az_publication_abstract field.field.node.az_publication.field_az_publication_approximate field.field.node.az_publication.field_az_publication_category field.field.node.az_publication.field_az_publication_container field.field.node.az_publication.field_az_publication_date field.field.node.az_publication.field_az_publication_date_type field.field.node.az_publication.field_az_publication_doi field.field.node.az_publication.field_az_publication_extra_links field.field.node.az_publication.field_az_publication_image field.field.node.az_publication.field_az_publication_issue field.field.node.az_publication.field_az_publication_link field.field.node.az_publication.field_az_publication_location field.field.node.az_publication.field_az_publication_media field.field.node.az_publication.field_az_publication_page field.field.node.az_publication.field_az_publication_pmid field.field.node.az_publication.field_az_publication_publisher field.field.node.az_publication.field_az_publication_type field.field.node.az_publication.field_az_publication_version field.field.node.az_publication.field_az_publication_volume field.storage.az_author.field_az_author_drop_particle field.storage.az_author.field_az_author_fname field.storage.az_author.field_az_author_link field.storage.az_author.field_az_author_literal field.storage.az_author.field_az_author_lname field.storage.az_author.field_az_author_nondrop_particle field.storage.az_author.field_az_author_parse field.storage.az_author.field_az_author_person field.storage.az_author.field_az_author_suffix field.storage.node.field_az_accessed_date field.storage.node.field_az_authors field.storage.node.field_az_publication_abstract field.storage.node.field_az_publication_approximate field.storage.node.field_az_publication_category field.storage.node.field_az_publication_container field.storage.node.field_az_publication_date field.storage.node.field_az_publication_date_type field.storage.node.field_az_publication_doi field.storage.node.field_az_publication_extra_links field.storage.node.field_az_publication_image field.storage.node.field_az_publication_issue field.storage.node.field_az_publication_link field.storage.node.field_az_publication_location field.storage.node.field_az_publication_media field.storage.node.field_az_publication_page field.storage.node.field_az_publication_pmid field.storage.node.field_az_publication_publisher field.storage.node.field_az_publication_type field.storage.node.field_az_publication_version field.storage.node.field_az_publication_volume node.type.az_publication pathauto.pattern.az_publication_url_pattern taxonomy.vocabulary.az_publication_categories views.view.az_publications views.view.az_publications_curate
lando drush pm:install -y az_publication
```

Or via `db-export` and `db-import`

```
lando db-export [file] 
lando drush en -y az_publication
lando db-import <file>
```
Anyway, posting here for further evaluation by the group.

## Related issues
Closes #3033

## How to test
Navigate to the Extension installation page as an admin
Install Quickstart publications
Navigate to the Extension Uninstall page as an admin
Uninstall Quickstart publications

See no errors.

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
